File: server/powercor_dashboard/dashboard/serializers/project.py
from core.serializers.user import ProjectUserSerializer
from dashboard.models.project import Project

from dashboard.serializers.discipline import DisciplineSerializer, DisciplineInListSerializer
from drf_writable_nested import WritableNestedModelSerializer


class ProjectInListSerializer(WritableNestedModelSerializer):
    disciplines = DisciplineInListSerializer(many=True)
    manager = ProjectUserSerializer()

    class Meta:
        model = Project
        fields = (
            'network',
            'name',
            'manager',
            'client',
            'comment',
            'disciplines',
            'business_importance',
            'priority',
            'status',
        )
        read_only_fields = ('priority',)


class ProjectSerializer(ProjectInListSerializer):
    disciplines = DisciplineSerializer(many=True, partial=True)
    manager = ProjectUserSerializer()

    class Meta(ProjectInListSerializer.Meta):
        pass

File: client/web/src/actions/project.js
import store from '@store'
import { Project } from '@store/project'


export const setProjects = (data) => {
    data.map(obj => {
        let project = Project.create({
            network: `${obj.network}`,
            name: obj.name,
            manager: `${obj.manager}`,
            client: `${obj.client}`,
            comment: obj.comment,
            businessImportance: `${obj.businessImportance}`,
            priority: obj.priority,
            status: obj.status,
        })
        store.projectStore.addProject(project)
    })
}